src/ConnectedMode.UnitTests/Persistence/BindingDtoSerializationTests.cs

```diff
@@ -0,0 +1,211 @@
+                "Name": null
+              },
+              "ProjectKey": "my_project_123",
+              "Profiles": {
+                "C": {
+                  "ProfileKey": "qpkey",
+                  "ProfileTimestamp": "2020-12-31T23:59:59"
+                }
+              }
+            }
+            """);
+    }
+    
+    [TestMethod]
+    public void Dto_FromSonarQubeBinding_SerializedAsExpected()
+    {
+        var serializeObject = JsonConvert.SerializeObject(bindingDtoConverter.ConvertToDto(boundSonarQubeServerProject), Formatting.Indented);
+
+        serializeObject.Should().BeEquivalentTo(
+            """
+            {
+              "ServerConnectionId": "http://next.sonarqube.com/sonarqube",
+              "ServerUri": "http://next.sonarqube.com/sonarqube",
+              "ProjectKey": "my_project_123",
+              "Profiles": {
+                "C": {
+                  "ProfileKey": "qpkey",
+                  "ProfileTimestamp": "2020-12-31T23:59:59"
+                }
+              }
+            }
+            """);
+    }
+    
+    [TestMethod]
+    public void Legacy_ToJson_ToDto_ToLegacy_IsCorrect()
+    {
+        var serialized = JsonConvert.SerializeObject(boundSonarQubeProject);
+        var convertedFromDtoToLegacy = bindingDtoConverter.ConvertFromDtoToLegacy(JsonConvert.DeserializeObject<BindingDto>(serialized), null);
+        
+        convertedFromDtoToLegacy.Should().BeEquivalentTo(boundSonarQubeProject);
+    }
+    
+    [TestMethod]
+    public void Legacy_ToJson_ToLegacyDirectly_And_ToDto_ToLegacy_IsCorrect()
+    {
+        var serialized = JsonConvert.SerializeObject(boundSonarQubeProject);
+        var legacyDirect = JsonConvert.DeserializeObject<BoundSonarQubeProject>(serialized);
+        var convertedFromDtoToLegacy = bindingDtoConverter.ConvertFromDtoToLegacy(JsonConvert.DeserializeObject<BindingDto>(serialized), null);
+        
+        convertedFromDtoToLegacy.Should().BeEquivalentTo(legacyDirect);
+    }
+    
+    [TestMethod]
+    public void Dto_ToJson_ToLegacy_IsCorrect()
+    {
+        var convertedFromDtoToLegacy = JsonConvert.DeserializeObject<BoundSonarQubeProject>(JsonConvert.SerializeObject(bindingDto));
+        
+        convertedFromDtoToLegacy.Should().BeEquivalentTo(boundSonarQubeProject);
+    }
+    
+    [TestMethod]
+    public void Dto_ToJson_ToDto_ToLegacy_IsCorrect()
+    {
+        var convertedFromDtoToLegacy = bindingDtoConverter.ConvertFromDtoToLegacy(JsonConvert.DeserializeObject<BindingDto>(JsonConvert.SerializeObject(bindingDto)), null);
+        
+        convertedFromDtoToLegacy.Should().BeEquivalentTo(boundSonarQubeProject);
+    }
+    
+    [TestMethod]
+    public void CurrentSonarCloud_ToDto_ToJson_ToDto_ToLegacy_IsCorrect()
+    {
+        var legacyBinding = bindingDtoConverter.ConvertFromDtoToLegacy(JsonConvert.DeserializeObject<BindingDto>(JsonConvert.SerializeObject(bindingDtoConverter.ConvertToDto(boundSonarCloudServerProject))), null);
```

Comment:
Please extract all the train wrecks into multiple variables, to make the test more readable.